openapi.yaml

```diff
@@ -12,106 +12,458 @@ info:
-          description: Filter results by age
+          name: filter[genre]
+          description: Filter books by genre.
+          required: false
+          schema:
+            type: string
+        - in: query
+          name: filter[description]
+          description: Search by book description.
+          required: false
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/BookSetResult'
+        '400':
+          $ref: '#/components/responses/400'
+        '500':
+          $ref: '#/components/responses/500'
+    post:
+      summary: Add a new book to library
+      tags:
+        - books
+      operationId: postBook
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/BookPostBody'
+      responses:
+        '201':
+          description: Book successfully created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BookResult'
+        '400':
+          $ref: '#/components/responses/400'
+        '409':
+          $ref: '#/components/responses/409'
+        '500':
+          $ref: '#/components/responses/500'
+  /books/{id}:
+    get:
+      summary: Retrieve details of a specific book.
+      tags:
+        - books
+      description: Get book by unique identifier.
+      operationId: getBookById
+      parameters:
+        - $ref: '#/components/parameters/bookId'
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BookResult'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+    patch:
+      summary: Update details of a specific book.
+      tags:
+        - books
+      description: Update a book by unique identifier.
+      operationId: updateBookById
+      parameters:
+        - $ref: '#/components/parameters/bookId'
+      requestBody:
+        required: true
+        description: patch body for a book
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/BookPatchBody'
+      responses:
+        '200':
+          description: Details of the updated book.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BookResult'
+        '400':
+          $ref: '#/components/responses/400'
+        '404':
+          $ref: '#/components/responses/404'
+        '409':
+          $ref: '#/components/responses/409'
+        '500':
+          $ref: '#/components/responses/500'
+    delete:
```

Comment:
Good. But let's not do DELETE(s) for your learning project.

Reply:
sure, should I remove delete from design then or let it be?

Reply:
Let's remove them from the PR.

Reply:
done, removed.